buffer/ring_growing.go

```diff
@@ -56,13 +51,8 @@ func (r *RingGrowing) WriteOne(data interface{}) {
 		// Time to grow
 		newN := r.n * 2
 		newData := make([]interface{}, newN)
-		to := r.beg + r.readable
-		if to <= r.n {
-			copy(newData, r.data[r.beg:to])
-		} else {
-			copied := copy(newData, r.data[r.beg:])
-			copy(newData[copied:], r.data[:(to%r.n)])
-		}
+		copied := copy(newData, r.data[r.beg:])
```

Comment:
This takes a common case only making a single copy and makes it into a function that always does two copies.  I'll admit those copies don't sometimes do a divide.

I'll take this if a benchmark test that shows for reasonable sizes this new implementation is cheaper.